modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/UserAccountResourceTest.java

```diff
@@ -1941,6 +1953,39 @@ private void _testPatchUserAccountWithImageExternalReferenceCode()
 		Assert.assertTrue(patchUserAccount.getImageId() > 0);
 	}
 
+	private void _testPostAccountUserAccountsByExternalReferenceCodeByEmailAddressWithRoleId()
+		throws Exception {
+
+		List<User> users = Arrays.asList(
+			UserTestUtil.addUser(), UserTestUtil.addUser(),
+			UserTestUtil.addUser(), UserTestUtil.addUser());
+
+		AccountRole accountRole = _accountRoleLocalService.addAccountRole(
+			RandomTestUtil.randomString(), TestPropsValues.getUserId(),
+			AccountConstants.ACCOUNT_ENTRY_ID_DEFAULT,
+			RandomTestUtil.randomString(), null,
+			Collections.singletonMap(
+				LocaleUtil.getDefault(), RandomTestUtil.randomString()));
+
+		for (User user : users) {
+			Assert.assertNull(
+				_accountEntryUserRelLocalService.fetchAccountEntryUserRel(
+					_accountEntry.getAccountEntryId(), user.getUserId()));
+		}
+
+		userAccountResource.
+			postAccountUserAccountsByExternalReferenceCodeByEmailAddress(
+				_accountEntry.getExternalReferenceCode(),
+				String.valueOf(accountRole.getAccountRoleId()),
+				_toEmailAddresses(users));
+
+		for (User user : users) {
+			Assert.assertNotNull(
```

Comment:
we just this check I don't think we are also checking if the role we are passing is applied. Can you please check?